kusto/data/errors/errors.go

```diff
@@ -165,9 +165,9 @@ func (e *Error) Error() string {
 	return b.String()
 }
 
-// Retry determines if the error is transient and the action can be retried or not.
+// ShouldRetry determines if the error is transient and the action can be retried or not.
 // Some errors that can be retried, such as a timeout, may never succeed, so avoid infinite retries.
-func Retry(err error) bool {
+func ShouldRetry(err error) bool {
```

Comment:
While I get your reasoning, the case for making the change is disrupted by the additional function calls that are now add confusion makes the API worse than the cure.

Please revert the name change.